packages/react/src/components/Storage/FileUploader/FileUploader.tsx

```diff
@@ -29,25 +29,50 @@ export function FileUploader({
     onDragStart,
     onDrop,
     setShowPreviewer,
-    setTargetFiles,
+    addTargetFiles,
     showPreviewer,
+    setFiles,
   } = useFileUploader();
+  const [allFileNames, setAllFileNames] = useState<string[]>([]);
 
   useEffect(() => {
     setShowPreviewer(isPreviewerVisible);
   }, [setShowPreviewer, isPreviewerVisible]);
 
+  useEffect(() => {
+    setAllFileNames(files.map((file) => file.name));
+  }, [files]);
+
+  // Previewer Methods
+
   const onFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    if (!event.target.files || event.target.files.length === 0) return;
+    if (!event.target.files || event.target.files.length === 0) {
+      return;
```

Comment:
I moved all these methods up from the Previewer to this FileUploader component. It will mean more prop drilling, but it does make it so our Previewer and Tracker components have less logic in them, and are more presentational